src/globus_cli/login_manager/tokenstore.py

```diff
@@ -116,6 +122,9 @@ def internal_auth_client():
         client_secret = client_data["client_secret"]
     else:
         # register a new instance client with auth
+        if is_client_login():
+            raise ValueError("client logins shouldn't create internal auth clients")
```

Comment:
What about moving this check to the start of the method?

I see that there's handling for this in the `delete_templated_client` method, and it might be simpler if we assert that this method can _never_ run when `is_client_login() == True`.